bridge/pkg/readiness/health.go

```diff
@@ -0,0 +1,68 @@
+// package readiness implements a minimal health-checking mechanism for use as k8s readiness probes. It will always
+// return a "ready" state after the conditions have been met for the first time - it's not meant for monitoring.
+//
+// Uses a global singleton registry (similar to the Prometheus client's default behavior).
+package readiness
+
+import (
+	"bytes"
+	"fmt"
+	"net/http"
+	"sync"
+)
+
+var (
+	mu       = sync.Mutex{}
+	registry = map[string]bool{}
+)
+
+// RegisterComponent registers the given component name such that it is required to be ready for the global check to succeed.
+func RegisterComponent(component string) {
```

Comment:
I think it would be better to use a type alias `ReadinessComponent` and specify them as constants within the app.

This makes searching easier and removes the magic strings in various places.